### DISCUSSION TITLE
versions of  prism 7 and 8 exception on contentcontrol region for framework 4.7

### DISCUSSION THREAD

**Discussion by stevensrf11** [PrismLibrary/Prism#2383] [WPF]
Does a region adapter now have to be created for a ContentContol?
I created  WPF .Net Framework 4.7 Application identical to the Sample 2 Region demo. I tested every version of Prism 7  annd Prism 8, and for all version an exception is thown about a region adapter not being create for content control.
Has anyone else heard reports on this happening from anything ?

The eception occurs here in Prim's code
public static void UpdateRegions()
        {

            try
            {
                updatingRegionsListeners.Raise(null, EventArgs.Empty);
            }
            catch (TargetInvocationException ex)
            {
                Exception rootException = ex.GetRootException();

                throw new UpdateRegionsException(string.Format(CultureInfo.CurrentCulture,
                    Resources.UpdateRegionException, rootException), ex.InnerException);
            }
        }

**Comment by stevensrf11** [comment]
Well I found the problem for the .Net Framewrok,
If the Project  has a period in it the the exception occurs.
For Example name a WPF Project  Wpf.SampleApp the add prism unity and give a region name to a content control and the exception occurs. That sucks.

**Comment by stevensrf11** [comment]
Well I found the previous email about the fix may be incorrect.
It  seems if the ContentControl Content is set like below the exception is thrown. Like Shown below
<Window x:Class="WpfApp1.Views.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:prism="http://prismlibrary.com/"
        Title="Shell" Height="350" Width="525">
    <Grid>
        <ContentControl prism:RegionManager.RegionName="ContentRegion" Content="Shell"/>
    </Grid>
</Window>
If the ContentControl Content is not shown at all as shown below the exception does not occurr
<Window x:Class="WpfApp1.Views.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:prism="http://prismlibrary.com/"
        Title="Shell" Height="350" Width="525">
    <Grid>
        <ContentControl prism:RegionManager.RegionName="ContentRegion" />
    </Grid>
</Window>

**Comment by brianlagunas** [comment]
See #2389
